setup.py

```diff
@@ -12,7 +12,18 @@
     author_email='kmike84@gmail.com',
     url='https://github.com/scrapinghub/scrapy-poet',
     packages=find_packages(exclude=['tests', 'example']),
-    install_requires=['andi>=0.3', 'attrs', 'parsel', 'web-poet'],
+    install_requires=[
+        # We need to install pyasn1 before trying to install service_identity,
+        # one of Scrapy's dependencies. This is a bug related to setuptools.
+        #
+        # https://github.com/pypa/setuptools/issues/498
+        'pyasn1',
+        'andi>=0.3',
+        'attrs',
+        'parsel',
+        'scrapy>=2.1.0',
```

Comment:
Hm, I'm not sure about this change. On one hand, it seems some framework-related libraries do that (e.g. some django-related packages do require django in setup.py). On the other hand, not everyone is doing this, and adding a framework as an install_requirement for a plugin can lead to problems - e.g. a framework can be occasionally upgraded when user tries to upgrade a package. 

pip used to upgrade all dependencies always, when upgrade of a package is requested; this seems to be changed (now it upgrades only dependencies which don't have a compatible version installed), but it still may lead to problems. Why would user want to have Scrapy upgraded (or installed) when `pip install scrapy-poet` or `pip install -U scrapy-poet` is executed? For me failing in case of incompatible Scrapy version looks like a better behavior than silently upgrading Scrapy.

An issue with docs building and CI can be solved in different ways, e.g. having docs/requirements.txt file which includes Scrapy.

Reply:
Thank you for your comments. That makes sense. I've reverted the setup.py file and included a requirements.txt to the docs directory. Doc is  still building fine after that :)